### ISSUE TITLE
Add line mark capabilities like interpolate to trail mark

### ISSUE THREAD

**Issue by nyurik** [vega/vega#1270]
Trail & line marks could really benefit from each other's capabilities.  Line allows interpolation, while trail allows different thickness per node.  It would be great to join these two capabilities in the same mark, allowing complex graphs like sankey ([good Google example](https://developers.google.com/chart/interactive/docs/gallery/sankey)).  I guess the only fundamental difference between line & trail is how fill behaves.

**Comment by jheer** [comment]
I agree this could be nice, but would require extensive graphics programming. We heavily rely on d3-shape for line and area marks, including all the interpolated curves. If someone wishes to (re-)implement this library with support for variable-width paths, then it might be possible to combine features with trail marks. For the meantime, however, we're unlikely to make changes here, with trail supporting only linear segments of variable width.

**Comment by tonne1** [comment]
Could the [d3-sankey module](https://github.com/d3/d3-sankey) be of any use for vega?
